lib/improver/units.py

```diff
@@ -38,28 +38,34 @@
 
 "unit": <str>
     The standard/default units for the coordinate or diagnostic
-    described by the key.
+    described by the key.  This is mandatory.
 "dtype": <dtype>
     The standard/default data type in which the coordinate points
-    or diagnostic values should be stored.
+    or diagnostic values should be stored.  This is optional; if
+    not set, float32 is assumed.
 """
 
 import numpy as np
 
 DEFAULT_UNITS = {
+    # time coordinates and suitable substrings
     "time": {
         "unit": "seconds since 1970-01-01 00:00:00",
         "dtype": np.int64},
-    "forecast_reference_time": {
-        "unit": "seconds since 1970-01-01 00:00:00",
-        "dtype": np.int64},
     "forecast_period": {
         "unit": "seconds",
         "dtype": np.int32},
-    "lwe_thickness_of_precipitation_amount": {
-        "unit": "m",
-        "dtype": np.float32},
-    "lwe_precipitation_rate": {
-        "unit": "m s-1",
-        "dtype": np.float32},
+    # other standard coordinates and substrings
+    "longitude": {"unit": "degrees"},
+    "latitude": {"unit": "degrees"},
+    "projection_x_coordinate": {"unit": "m"},
+    "projection_y_coordinate": {"unit": "m"},
+    "percentile": {"unit": "%"},
+    "probability": {"unit": "1"},
+    # standard diagnostics and suitable substrings
+    "temperature": {"unit": "K"},
```

Comment:
Perhaps the keys in this dictionary should be alphabetised to make it easier to read? What do you think?

Reply:
I'm halfway with you ... given how this will extend I think you're right about the list of diagnostics.  But I think it's a good idea to keep the dictionary categorised as I have with comment strings, rather than ordering all entries strictly alphabetically.